src/PolykeyClient.ts

```diff
@@ -57,20 +67,26 @@ class PolykeyClient {
         logger: logger.getChild(Session.name),
         fresh,
       }));
-    grpcClient =
-      grpcClient ??
-      (await GRPCClientClient.createGRPCClientClient({
-        nodeId,
-        host,
-        port,
-        tlsConfig: { keyPrivatePem: undefined, certChainPem: undefined },
-        session,
-        timer,
-        logger: logger.getChild(GRPCClientClient.name),
-      }));
+    const webSocketClient = await WebSocketClient.createWebSocketClient({
+      host,
+      port,
+      expectedNodeIds: [nodeId],
+      logger: logger.getChild(WebSocketClient.name),
+      connectionTimeout,
+      pingTimeout,
+    });
```

Comment:
That slipped my mind at the time. At the time I had a problem with providing the `RPCClient` as a parameter, I don't recall exactly why. I'll have the review and fix this because as you mention, this needs to be agnostic to the underlying transport which can be `Quic` or `WebSocket`.

An oversight on my part.

Reply:
This is now fixed, Instead of the transport being internal to the `PolykeyClient`it now takes the `StreamFactory` callback. transport is now created and managed separately.